### DISCUSSION TITLE
Additional cost example

### DISCUSSION THREAD

**Discussion by morlic** [custom-components/nordpool#354] [Q&A] [answered]
Hi,

I'm trying to use additional costs template to calculate the price that I will pay in the end. The price structure is roughly:

`( current_price * FACTOR + OFFSET ) * VAT`

I see from the code that the integration calculates VAT (if set to true) before current_prices is passed to the template, which I can adjust for, but I still get values which are too high - for example for current_price = 1, FACTOR of 1.30, offset of 0.67, I was expecting to get 2.3 (VAT is 1.25) but instead the result was around 4.

Any suggestions how to set the template to get this right? I know I could create an additional template sensor in Hass to do that, but since this is already in place... :)

Thanks,
 Marin

**Comment by morlic** [comment, marked as answer]
I reread the section on Additional costs, what I thought should be:

`{{ ( current_price * 1.30 + 0.67 ) | float }}` is actually `{{ current_price * 0.30 + 0.67 * 1.25 | float }}`

additional_cost is added on top of the current_price, does not replace it completely as I thought.

**Comment by phiwiko** [comment]
i try to integrate seasonal additional costs with vary periods a day and dont get it right.

for expample:
in quarter 1 there are net fees 
from 0-5 @ 0,0076 €/kWh
from 5-9 @ 0,0762 €/kWh
from 9-14 @ 0,99 €/kWh

etc.
how do i store these values in the stored values of the whole forecast-period?

**Reply by sillymoi** [reply]
maybe store these in a separate helper or schedule and then read it from there (state attribute)?

**Reply by phiwiko** [reply]
But then these costs would not be in the forecast, right?

I am able to sum up two states, but not to sum up to 96 value arrays. 

BTW. I am not able to make an array of these additional costs.

**Reply by sillymoi** [reply]
not sure, but there seems to be a template variable `now` that might contain the time of the sample, so that would cover the forecast, but i don't know if it's possible to use that with a schedule or you need to hardcode the fees.

https://github.com/custom-components/nordpool/blob/master/custom_components/nordpool/sensor.py#L270

**Comment by Hellowlol** [comment]
Check the readme example 5
